src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -1565,7 +1565,13 @@ public function newQueryForRestoration($ids)
      */
     public function newEloquentBuilder($query)
     {
-        return new Builder($query);
+        $returnType = (new \ReflectionMethod(static::class, 'query'))->getReturnType()?->getName();
```

Comment:
Can you import this class with a `use` statement?

Reply:
Sure, done